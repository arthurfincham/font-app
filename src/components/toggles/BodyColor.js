import ToggleButton from '@mui/material/ToggleButton';
import { useState } from 'react';
import FormatColorFillIcon from '@mui/icons-material/FormatColorFill';
import Collapse from '@mui/material/Collapse';
import FormControlLabel from '@mui/material/FormControlLabel';
import Box from '@mui/material/Box';
import { MyToggleButtonGroup, MyToggleButton, MyCircle } from '../styled/styledTheme';

export default function BodyColor({ setBodyColor, bodyColor }) {
  const themeColors = ['#FFF', '#FEF3C7', '#272C35', '#6366F2', '#F23C1F', '#D4D1FF', '#FA8C1E'];

  const [checked, setChecked] = useState(false);

  const handleCheck = () => {
    setChecked((prev) => !prev);
  };

  const handleColor = (event, newColor) => {
    setBodyColor(newColor);
  };

  return (
    <Box sx={{ width: '50px', position: 'relative', marginInline: '5px' }}>
      <FormControlLabel
        control={
          <ToggleButton checked={checked} onChange={handleCheck}>
            <FormatColorFillIcon htmlColor={bodyColor} />
          </ToggleButton>
        }
        label=""
        sx={{ margin: '0 auto' }}
      />
      <Collapse in={checked} collapsedSize={0} sx={{ position: 'absolute', right: '-5px', top: '70px' }}>
        <MyToggleButtonGroup size="small" value={bodyColor} exclusive onChange={handleColor} aria-label="site theme" orientation="vertical">
          {themeColors.map((color) => {
            return (
              <MyToggleButton value={color} aria-label="left aligned" sx={{ zIndex: 10 }}>
                <MyCircle color={color} />
              </MyToggleButton>
            );
          })}
        </MyToggleButtonGroup>
      </Collapse>
    </Box>
  );
}
