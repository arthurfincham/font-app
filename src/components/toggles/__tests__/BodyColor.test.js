import BodyColor from '../BodyColor';
import React from 'react';
import Enzyme, { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import 'jest-styled-components';

Enzyme.configure({ adapter: new Adapter() });

it('it correctly renders the BodyColor toggle', () => {
  const wrapper = shallow(<BodyColor />);

  expect(toJson(wrapper)).toMatchSnapshot();
});
