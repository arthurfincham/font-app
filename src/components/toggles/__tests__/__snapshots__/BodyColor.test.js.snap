// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it correctly renders the BodyColor toggle 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "marginInline": "5px",
      "position": "relative",
      "width": "50px",
    }
  }
>
  <ForwardRef(FormControlLabel)
    control={
      <ForwardRef(ToggleButton)
        onChange={[Function]}
        value={false}
      >
        <Memo />
      </ForwardRef(ToggleButton)>
    }
    label=""
    sx={
      Object {
        "margin": "0 auto",
      }
    }
  />
  <ForwardRef(Collapse)
    collapsedSize={0}
    in={false}
    sx={
      Object {
        "position": "absolute",
        "right": "-5px",
        "top": "70px",
      }
    }
  >
    <Styled(ForwardRef(ToggleButtonGroup))
      aria-label="site theme"
      exclusive={true}
      onChange={[Function]}
      orientation="vertical"
      size="small"
    >
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#FFF"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#FFF"
      >
        <Styled(memo)
          color="#FFF"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#FEF3C7"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#FEF3C7"
      >
        <Styled(memo)
          color="#FEF3C7"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#272C35"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#272C35"
      >
        <Styled(memo)
          color="#272C35"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#6366F2"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#6366F2"
      >
        <Styled(memo)
          color="#6366F2"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#F23C1F"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#F23C1F"
      >
        <Styled(memo)
          color="#F23C1F"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#D4D1FF"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#D4D1FF"
      >
        <Styled(memo)
          color="#D4D1FF"
        />
      </Styled(ForwardRef(ToggleButton))>
      <Styled(ForwardRef(ToggleButton))
        aria-label="left aligned"
        key="#FA8C1E"
        sx={
          Object {
            "zIndex": 10,
          }
        }
        value="#FA8C1E"
      >
        <Styled(memo)
          color="#FA8C1E"
        />
      </Styled(ForwardRef(ToggleButton))>
    </Styled(ForwardRef(ToggleButtonGroup))>
  </ForwardRef(Collapse)>
</ForwardRef(Box)>
`;
